plugins/chat/app/controllers/api/hints_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class Chat::Api::HintsController < ApplicationController
+  before_action :ensure_logged_in
+
+  def check_group_mentions
+    group_names = params[:mentions]
+
+    raise Discourse::InvalidParameters.new(:mentions) if group_names.blank?
+    RateLimiter.new(current_user, "group_mention_hints", 5, 10.seconds).performed!
+
+    all_groups = Group
+      .where("LOWER(name) IN (?)", group_names)
+      .visible_groups(current_user)
+      .pluck(:name)
+
+    mentionable_groups = filter_mentionable_groups(all_groups)
+
+    result = {
+      unreachable: all_groups - mentionable_groups.map(&:name),
+      over_members_limit: mentionable_groups.select { |g| g.user_count > SiteSetting.max_users_notified_per_group_mention }.map(&:name),
```

Comment:
`over_members_limit` is a little bit unclear to me, it sounds like the group has too man many members, when we only refuse to mention that many people.

Maybe just `over_mentions_limit` would be enough?

Reply:
> it sounds like the group has too man many members

Yes, that's actually what's happening here. This is not related to the number of mentions in the message but to the number of members the group has.